src/main/java/org/fcrepo/client/FcrepoResponse.java

```diff
@@ -55,6 +72,36 @@ public FcrepoResponse(final URI url, final int statusCode,
     }
 
     /**
+     * {@inheritDoc}
+     * <p>
+     * Implementation note: Invoking this method will close the underlying {@code InputStream} containing the entity
+     * body of the HTTP response.
+     * </p>
+     *
+     * @throws IOException if there is an error closing the underlying HTTP response stream.
+     */
+    @Override
+    public void close() throws IOException {
```

Comment:
While I agree that this work should be done, I'm convinced this should be handled in another Jira issue.  Primarily because of the amount of work involved: `setBody(InputStream)` on the `FcrepoReponse` is used a lot in the tests.  If the `InputStream body` member is going away in favor of a `CloseableHttpResponse` member, then `setBody(InputStream)` no longer makes sense to keep.  This is also tied into making `FcrepoResponse` immutable, which probably shouldn't be a part of this work either.

Secondarily, the aim of this PR is to really make it clear who is responsible for resource management, and that's been accomplished.  Using `CloseableHttpResponse` in `FcrepoResponse` is a (worthy) optimization, as is an immutable `FcrepoResponse`, but I'm not sure they belong in this PR.